src/ui/public/agg_types/buckets/terms.js

```diff
@@ -73,6 +73,17 @@ export const termsBucketAgg = new BucketAggType({
     const params = agg.params;
     return agg.getFieldDisplayName() + ': ' + params.order.display;
   },
+  getFormat: function (bucket) {
+    return {
+      getConverterFor: (type) => {
+        return (val) => {
+          if (val === '__other__') return bucket.params.otherBucketLabel;
+          if (val === '__missing__') return bucket.params.missingBucketLabel;
+          return bucket.params.field.format.getConverterFor(type)(val);
```

Comment:
Slight performance improvement, move `bucket.params.field.format.getConverterFor(type)` into `gerConverterFor` above the `return`.